### DISCUSSION TITLE
Let `<TransitionChild>` inherit `unmount` from `<Transition>`

### DISCUSSION THREAD

**Discussion by ulken** [tailwindlabs/headlessui#799] [Ideas]
Using `show` on `<Transition>` to coordinate multiple `<Transition.Child>` transitions is great. However, I find it a little clumsy to have to define `unmount` on both `<Transition>` and `<Transition.Child>`.

Any chance `<Transition.Child>` could inherit `unmount` from `<Transition>`, similar to how `show` is treated? I guess it would be preferable to be able to override them as well.

Or are there any specific technical reasons preventing this?

---

[Edit]: `appear` seems to work as expected.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your suggestion!
Much appreciated! 🙏

I will have to come back to this one. There is no specific technical reason why it wouldn't work. I just have to decide if it has to be opt-in or opt-out. I don't want to break backwards compatibility though.

I have been thinking about a config wrapper where you can define defaults for a whole tree, something like this:

```js
<Configuration defaults={{ unmount: true, appear: false }}>
...
</Configuration>
```

I'll keep you posted on this one!

**Comment by ulken** [comment]
Yeah, I like that idea. Similar to how e.g.[SWR](https://swr.vercel.app/docs/global-configuration) does it.

However, I can still see benefits of having `<Transition.Child>` inheriting it from `<Transition>` too.
